File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.py
import math

def solution(n):
    answer = 0
    for i in range(2,n+1):
        if isPrime(i):
            answer+=1
    return answer

def isPrime(x):
    for i in range (2, int(math.sqrt(x)) + 1):
    	if x % i == 0:
        	return False
    return True

# def isPrime(n):
#     if n == 2:
#         return True
#     for i in range(2,n):
#         if n%i==0:
#             return False
#     return True

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>
#include <numeric>
#include <cmath>

using namespace std;

int solution(int n) {
    int answer = 0;
    vector<bool> v(n+1,true);
    v[0] = false;
    v[1] = false;
    for(int i = 2; i < int(sqrt(n))+1; i++){
        for(int j = i*2; j <= n; j+=i){
            v[j] = false;
        }
    }
    answer = accumulate(v.begin(), v.end(), 0);
    return answer;
}